content/articles/haskell-quickcheck-in-python.md

```diff
@@ -0,0 +1,91 @@
+Title: Using Haskell's QuickCheck from Python
+Date: 2015-03-19
+Short_summary: We show how Python testing can be done via Haskell's QuickCheck.
+Category: Hacks
+Authors: Tom
+
+Everyone I know who has ever used
+[QuickCheck](http://hackage.haskell.org/package/QuickCheck/) in anger
+cannot live without it.
+
+The basic idea behind QuickCheck is that we write a proposition, e.g.:
+*Reversing a list twice returns the same as the original list*. In
+code:
+
+```python
+reverse(reverse(x)) == x
+```
+
+QuickCheck will try to find a counterexample to that proposition by
+generating random values for `x`. And it is very good at finding mean values:
+
+<blockquote class="twitter-tweet" lang="en"><p>Haskell Quickcheck enters a bar: asks for 1 beer, 42 beers, -Inifinity beers, shaves bartenders beard, sets off a tactical nuke.</p>&mdash; Michael Neale (@michaelneale) <a href="https://twitter.com/michaelneale/status/567532684595851264">February 17, 2015</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
```

Comment:
This degrades gracefully like our commenting system. I.e. it will still render without js